### PR TITLE
Refs [TASK][36334] : Create Seting Screen

### PR DESCRIPTION
**Pull request by hideoncalm**
## Related Tickets
- [#Task 36334: Create Seting Screen](https://edu-redmine.sun-asterisk.vn/issues/36334)
- [#Task 36335: Handle event click button caculator](https://edu-redmine.sun-asterisk.vn/issues/36335)
- [#Task 36336: Click button to change language](https://edu-redmine.sun-asterisk.vn/issues/36336)

## WHAT
- 
## Evidence (Screenshot or Video)
image
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/coinqapp/utils/AppSharedPreferences.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+
+class AppSharedPreferences private constructor(context: Context) {
+
+    private val preferences = context.getSharedPreferences(SHARED_PREFERENCES_NAME, MODE_PRIVATE)
+    private val editor = preferences.edit()
+
+    fun writeString(key: String, value: String) {
```
Comment (tuannm-1276) [inline app/src/main/java/com/sunasterisk/coinqapp/utils/AppSharedPreferences.kt]: push gì?

File: app/src/main/java/com/sunasterisk/coinqapp/utils/AppSharedPreferences.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        editor.apply {
+            putString(key, value)
+            commit()
+        }
+    }
+
+    fun getString(key: String, defaultString: String) = preferences.getString(key, defaultString)
```
Comment (tuannm-1276) [inline app/src/main/java/com/sunasterisk/coinqapp/utils/AppSharedPreferences.kt:18]: get gì?

File: app/src/main/java/com/sunasterisk/coinqapp/utils/AppSharedPreferences.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+class AppSharedPreferences private constructor(context: Context) {
+
+    private val preferences = context.getSharedPreferences(SHARED_PREFERENCES_NAME, MODE_PRIVATE)
+    private val editor = preferences.edit()
+
+    fun putStringByKey(key: String, value: String) {
+        editor.apply {
+            putString(key, value)
+            commit()
```
Comment (tuannm-1276) [inline app/src/main/java/com/sunasterisk/coinqapp/utils/AppSharedPreferences.kt]: thay bằng apply() đi em
nếu k thì chạy nó từ một thread khác

File: app/src/main/java/com/sunasterisk/coinqapp/utils/AppSharedPreferences.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+
+class AppSharedPreferences private constructor(context: Context) {
+
+    private val preferences = context.getSharedPreferences(SHARED_PREFERENCES_NAME, MODE_PRIVATE)
+    private val editor = preferences.edit()
+
+    fun putStringByKey(key: String, value: String) {
```
Comment (tuannm-1276) [inline app/src/main/java/com/sunasterisk/coinqapp/utils/AppSharedPreferences.kt]: ý anh là putName hay gì đó. thôi cứ để lại putString nha

File: app/src/main/java/com/sunasterisk/coinqapp/utils/AppSharedPreferences.kt [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        editor.apply {
+            putString(key, value)
+            commit()
+        }
+    }
+
+    fun getStringByKey(key: String, defaultString: String) = preferences.getString(key, defaultString)
```
Comment (tuannm-1276) [inline app/src/main/java/com/sunasterisk/coinqapp/utils/AppSharedPreferences.kt]: để lại getString vậy

File: app/src/main/java/com/sunasterisk/coinqapp/utils/AppSharedPreferences.kt [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+    companion object {
+        const val SHARED_PREFERENCES_NAME = "shared"
```
Comment (tuannm-1276) [inline app/src/main/java/com/sunasterisk/coinqapp/utils/AppSharedPreferences.kt]: Key nên đặt tên để khỏi tránh trùng lặp, gồm cả chuỗi package nữa

File: app/src/main/java/com/sunasterisk/coinqapp/utils/AppSharedPreferences.kt [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        editor.apply {
+            putString(key, value)
+            apply()
+        }
+    }
+
+    fun getString(key: String, defaultString: String) = preferences.getString(key, defaultString)
+
+    companion object {
+        const val PREF_FILE_NAME = "com.sunasterisk.coinqapp.shared"
```
Comment (tuannm-1276) [inline app/src/main/java/com/sunasterisk/coinqapp/utils/AppSharedPreferences.kt]:
```suggestion
        const val PREF_FILE_NAME = "com.sunasterisk.coinqapp. PREF_FILE_NAME"
```

**Review by thonc-2135 — APPROVED** [review]

**Pull request merged**